src/js/smooth-scroll/smooth-scroll.js

```diff
@@ -269,18 +269,24 @@
 	var setHistory = function (options) {
 
 		// Make sure this should run
-		if (!history.replaceState || !options.updateURL || history.state) return;
+		if (!history.replaceState || !options.updateURL || (history.state && history.state.smoothScroll)) return;
 
 		// Get the hash to use
 		var hash = window.location.hash;
 		hash = hash ? hash : window.pageYOffset;
 
+		var historyState = {
+			smoothScroll: JSON.stringify(options),
+			anchor: hash ? hash : window.pageYOffset
+		};
+		if (history.state) {
```

Comment:
I understand the bit above, but can you explain to me why this is needed?

Reply:
Look at this screen shot below,this is a SPA which is using vue and vue router.

When I push a new route,history state will automatically set by vue router,if you just replace history state,vue router will not work correctly.

If you use react router or set some state by yourself,this has the same problem.

![image](https://user-images.githubusercontent.com/11729409/51648868-3df11300-1fbd-11e9-99c4-39583515c6ff.png)